### ISSUE TITLE
Provide example of packaging separate layers into Java container after deprecation of rules_docker

### ISSUE THREAD

**Issue by C0mbatwombat** [bazelbuild/rules_java#160]
(It feels awkward to ask this question here, but it is a follow-up of [ this suggestion over at rules_oci ](https://github.com/bazel-contrib/rules_oci/issues/416#issuecomment-1832829593))

As the maintainers have decided that the rules_oci repository should remain language agnostic, it is quite unclear where to go with questions for language-specific containers, so following @alexeagle 's advice I am trying over here. 

I would like to migrate from rules_docker to rules_oci, but the functionality to package depsets into separate layers has not been provided by the new rules ([_jar_app_layer_impl](https://github.com/bazelbuild/rules_docker/blob/8e70c6bcb584a15a8fd061ea489b933c0ff344ca/java/image.bzl#L178-L212) ). 

Would it be possible to provide similar functionality in a `.defs` file in this repository, and could someone help me out in porting it?

**Comment by SanjayVas** [comment]
Related logic from `rules_java` (once these parts are actually migrated to Starlark) that was replicated in `jar_app_layer` was handling runfiles and native libraries.

**Comment by C0mbatwombat** [comment]
Hi @SanjayVas thanks for your reply, do I understand correctly that you are planning to migrate the functionality to rules_java?

**Comment by SanjayVas** [comment]
> do I understand correctly that you are planning to migrate the functionality to rules_java?

No. I'm not a contributor to rules_java. I'm just sharing some context for why this likely needs to be solved in rules_java instead of downstream.
